packages/server/modules/core/domain/objects/types.ts

```diff
@@ -9,12 +9,14 @@ export type SpeckleObject = ObjectRecord
 /**
  * We preemptively serialize a couple of fields (usually knex does it)
  */
-export type InsertableSpeckleObject = OverrideProperties<
-  SetOptional<SpeckleObject, 'createdAt'>,
-  {
-    data: string
-    totalChildrenCountByDepth: string
-  }
+export type InsertableSpeckleObject = NullableKeysToOptional<
```

Comment:
Cause totalChildrenCount and totalChildrenCountByDepth are actually optional (nullable in DB)

Reply:
yep, they are. we probably don't need 'em even (@gjedlicska please sanity check this)

Reply:
all this typing wizardry usually fries my mind

i'd like to simplify this into a common params, insert model and query model 3 type definition instead of this, which took me 5 mins to understand, cause of all the jumping around between shared, typefest and our own type definitions